tests/Console/Command/StageCommandTest.php

```diff
@@ -30,22 +47,88 @@ public function testBasicConfiguration(): void
         $definition = $command->getDefinition();
         $arguments = $definition->getArguments();
         $options = $definition->getOptions();
+        $composerCommandArgument = $definition->getArgument('composer-command');
 
         self::assertSame('stage', $command->getName(), 'Set correct name.');
         self::assertSame([], $command->getAliases(), 'Set correct aliases.');
         self::assertNotEmpty($command->getDescription(), 'Set a description.');
-        self::assertSame([], array_keys($arguments), 'Set correct arguments.');
+        self::assertSame(['composer-command'], array_keys($arguments), 'Set correct arguments.');
         self::assertSame([], array_keys($options), 'Set correct options.');
+        self::assertNotEmpty($command->getUsages(), 'Set usages.');
+        self::assertNotEmpty($command->getHelp(), 'Set help.');
+
+        self::assertTrue($composerCommandArgument->isRequired(), 'Required Composer command option.');
+        self::assertTrue($composerCommandArgument->isArray(), 'Required Composer command option.');
+        self::assertNotEmpty($composerCommandArgument->getDescription(), "Description provided.");
     }
 
     /**
      * @covers ::execute
+     *
+     * @dataProvider providerBasicExecution
      */
-    public function testBasicExecution(): void
+    public function testBasicExecution($composerCommand, $stagingDir, $output): void
     {
-        $this->executeCommand();
+        $this->stager
+            ->stage($composerCommand, $stagingDir)
+            ->shouldBeCalledOnce()
+            ->willReturn($output);
+
+        $this->executeCommand([
+            'composer-command' => $composerCommand,
+            '--staging-dir' => $stagingDir,
+        ]);
 
-        self::assertSame('', $this->getDisplay(), 'Displayed correct output.');
+        self::assertSame($output, $this->getDisplay(), 'Displayed correct output.');
         self::assertSame(ExitCode::SUCCESS, $this->getStatusCode(), 'Returned correct status code.');
     }
+
+    public function providerBasicExecution(): array
+    {
+        return [
+            [
+                'composerCommand' => [self::INERT_COMMAND],
+                'stagingDir' => 'lorem/ipsum',
+                'output' => 'lorem'
+            ],
+            [
+                'composerCommand' => [
+                    'update',
+                    '--with-all-dependencies',
+                ],
+                'stagingDir' => 'dolor/sit',
+                'output' => 'ipsum'
+            ],
+        ];
+    }
+
+    /**
+     * @covers ::execute
+     *
+     * @dataProvider providerCommandFailure
+     */
+    public function testCommandFailure($exception, $message): void
+    {
+        $this->stager
+            ->stage(Argument::any(), Argument::any())
```

Comment:
Pro tip: you could use `->stage(Argument::cetera())` here if you don't really care about the type or number of arguments.

Reply:
Really?! That'll be so helpful! Thanks for telling me!